cdc/log.cc

```diff
@@ -840,6 +840,13 @@ utils::UUID generate_timeuuid(api::timestamp_type t) {
     return utils::UUID_gen::get_random_time_UUID_from_micros(t);
 }
 
+/**
+ * Returns whether we should generate cdc delta values (beyond keys)
+ */
+static bool generate_delta_values(const schema& s) {
+    return s.cdc_options().get_delta_mode() == cdc::delta_mode::full;
```

Comment:
I think this should be decided once when builder is created and not checked every time we want to add something to the result

Reply:
Or when the visitor is created if we decide not to hide checks inside the builder.

Reply:
I find it hard to believe that an if per value is gonna cost more than for example even more templating -> code bloat -> icache misses and flushes. 

If a profile really shows that the checks adds measurable cost, we can consider making it embedded (templated), but until then I think that is a premature opt.

Reply:
What I mean is that the visitor (or the builder) should have `bool add_full_deltas = s.cdc_options().get_delta_mode() == cdc::delta_mode::full` set in the constructor and then the checks should be just `if (add_full_deltas) { ...`

Reply:
`const bool` even! Good for readability.

Reply:
Right, ok, agree. Fixing.